dashboard/modules/job/job_agent.py

```diff
@@ -0,0 +1,303 @@
+import asyncio
+import json
+import logging
+import os.path
+import itertools
+import subprocess
+import sys
+import uuid
+import traceback
+from abc import abstractmethod
+from typing import Union
+
+import attr
+from attr.validators import instance_of
+import ray.new_dashboard.utils as dashboard_utils
+from ray.new_dashboard.utils import create_task
+from ray.new_dashboard.modules.job import job_consts
+from ray.new_dashboard.modules.job.job_description import JobDescription
+from ray.core.generated import job_agent_pb2
+from ray.core.generated import job_agent_pb2_grpc
+from ray.core.generated import agent_manager_pb2
+from ray._private.utils import hex_to_binary, binary_to_hex
+
+logger = logging.getLogger(__name__)
+
+
+@attr.s(kw_only=True, slots=True)
+class JobInfo(JobDescription):
+    # The job id hex string.
+    job_id = attr.ib(type=str, validator=instance_of(str))
+    # The temp directory.
+    temp_dir = attr.ib(type=str, validator=instance_of(str))
+    # The log directory.
+    log_dir = attr.ib(type=str, validator=instance_of(str))
+    # The driver process instance.
+    driver = attr.ib(
+        type=Union[None, asyncio.subprocess.Process], default=None)
+
+    def __attrs_post_init__(self):
+        job_package_dir = job_consts.JOB_UNPACK_DIR.format(
+            temp_dir=self.temp_dir, job_id=self.job_id)
+        # Support json values for env.
+        self.env = {
+            k: v if isinstance(v, str) else json.dumps(v)
+            for k, v in self.env.items()
+        }
+        self.env["RAY_JOB_DIR"] = job_package_dir
```

Comment:
What is this used for? I don't see it elsewhere in the code. Leave a comment please.

Reply:
I have removed the `RAY_JOB_DIR`.